docs/jwst/datamodels/fits_files.rst

```diff
@@ -1,214 +1,600 @@
-|  2  | Data quality   | DQ      | IMAGE    | uint32    | ncols x nrows x nints |
-+-----+----------------+---------+----------+-----------+-----------------------+
-|  3  | Error values   | ERR     | IMAGE    | float32   | ncols x nrows x nints |
-+-----+----------------+---------+----------+-----------+-----------------------+
-
-The FITS file structure for a `cal` product is as follows:
-
-+-----+----------------+---------+----------+-----------+---------------+
-| HDU | Content        | EXTNAME | HDU Type | Data Type | Dimensions    |
-+=====+================+=========+==========+===========+===============+
-|  0  | Primary header | N/A     | N/A      | N/A       | N/A           |
-+-----+----------------+---------+----------+-----------+---------------+
-|  1  | Pixel values   | SCI     | IMAGE    | float32   | ncols x nrows |
-+-----+----------------+---------+----------+-----------+---------------+
-|  2  | Data quality   | DQ      | IMAGE    | uint32    | ncols x nrows |
-+-----+----------------+---------+----------+-----------+---------------+
-|  3  | Error values   | ERR     | IMAGE    | float32   | ncols x nrows |
-+-----+----------------+---------+----------+-----------+---------------+
+The following sections describe the format and contents of each of the JWST FITS science
+products. Things to note in the descriptions include:
+
+ - Not all FITS extensions occur in every data product of a given type. Many are either
+   optional or dependent on the instrument or observing mode. Such optional extensions are
+   noted with an asterisk in the tables below.
+
+ - Because some extensions are optional, as well as the fact that the exact ordering of the
+   extensions is not guaranteed, the FITS HDU index numbers of a given extension type can
+   vary from one product to another. The only guarantee is that the ``SCI`` extension,
+   containing the actual pixel values, will always be the first FITS extension (HDU=1).
+   Other common extensions, like ``DQ`` and ``ERR``, usually immediately follow the ``SCI``,
+   but the order is not guaranteed. Hence HDU index numbers are not listed for many
+   extension types, because they can vary.
+
+Uncalibrated raw data: ``uncal``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Exposure raw data products are designated by a file name
+suffix of "uncal." These files usually contain only the raw detector pixel values
+from an exposure, with the addition of some table extensions containing various types of
+meta data associated with the exposure.
+Additional extensions can be included for certain instruments and readout types, as noted
+below.
+The FITS file structure is as follows.
+
++-----+------------+----------+-----------+---------------------------------+
+| HDU | EXTNAME    | HDU Type | Data Type | Dimensions                      |
++=====+============+==========+===========+=================================+
+|  0  | N/A        | primary  | N/A       | N/A                             |
```

Comment:
No, the primary header data unit is not an extension.  All HDUs are header data units, but only the ones following the primary HDU are extensions.

Reply:
And the EXTNAME keyword can only appear in extension headers, not primary headers, so I didn't want to put the label "primary" in the EXTNAME column.

Reply:
Fair argument -- I was mainly thinking of name rather than EXTNAME and that `x['PRIMARY']` does work to return the HDU, but I would agree that it doesn't have an EXTNAME keyword and so my comment is not technically correct